src/transaction/log_global.c

```diff
@@ -141,15 +140,39 @@ log_global::wait_for_ps_flushed_lsa (const log_lsa &flush_lsa)
   quorum_consenesus_er_log ("Quorum satisfied: page server flushed LSA = %lld|%d.\n", LSA_AS_ARGS (&log_Gl.m_ps_consensus_flushed_lsa));
 }
 
+void
+log_global::initialize_log_prior_sender ()
+{
+  assert (m_prior_sender == nullptr);
+  m_prior_sender = std::make_unique<cublog::prior_sender> ();
+}
+
+void
+log_global::finalize_log_prior_sender ()
+{
+  assert (m_prior_sender != nullptr);
+  assert (m_prior_sender->is_empty ());
```

Comment:
It says it's empty, but something may be added before the next line, doesn't it?

Reply:
Indeed. However it is a far-fetched scenario.

I adapted: moved the assert in the prior sender dtor; removed from here. And adapted unit tests.